### PR TITLE
Simplification of concurrent data types

### PR DESCRIPTION
**Pull request by khajavi**
Addresses #6632

### CODE REVIEW

**Comment by khajavi** [comment]
@adamgfraser 
It's ready for your review.

File: docs/howto/migrate/migration-guide.md [hunk 1]
```diff
@@ -1537,8 +1551,8 @@ As the `RefM` is renamed to `Ref.Synchronized`; now the `Synchronized` is a subt
 
 To perform the migration, after renaming these types to the newer ones (e.g. `RefM` renamed to `Ref.Synchronized`) we should perform the following method renames:
 
-| ZIO 1.x                  | ZIO 2.x                                 |
-|--------------------------|-----------------------------------------|
+| ZIO 1.x                  | ZIO 2.x                               |
+|--------------------------|---------------------------------------|
 | `RefM#dequeueRef`       | `Ref.Synchronized#SubscriptionRef`     |
```
Comment (adamgfraser) [inline docs/howto/migrate/migration-guide.md:1556]: `SubscriptionRef` is in ZIO Stream.

**Review by adamgfraser — CHANGES_REQUESTED** [review]
One minor comment. Otherwise looks great!

File: docs/howto/migrate/migration-guide.md (hunk 1 above)
Comment (khajavi) [reply docs/howto/migrate/migration-guide.md]: fixed that!

**Review by adamgfraser — APPROVED** [review]

**Pull request merged**